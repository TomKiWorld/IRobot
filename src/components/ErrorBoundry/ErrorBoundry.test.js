import { shallow } from 'enzyme';
import React from 'react';
import ErrorBoundry from './ErrorBoundry';

describe('ErrorBoundry Component', () => {
  const Child = () => <h1>Test</h1>;
  const wrapper = shallow(<ErrorBoundry><Child /></ErrorBoundry>);

  it('Expect to render ErrorBoundry Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Expect to render Error message', () => {
    const error = new Error('hi!');
    wrapper.find(Child).simulateError(error);
    expect(wrapper.state().hasError).toBe(true);
  });
});
