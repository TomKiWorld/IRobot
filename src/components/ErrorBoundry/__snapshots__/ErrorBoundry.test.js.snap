// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundry Component Expect to render ErrorBoundry Component 1`] = `ShallowWrapper {}`;
